app/services/foreman_discovery/import_hooks/subnet_and_taxonomy.rb

```diff
@@ -4,35 +4,48 @@ module ImportHooks
     class SubnetAndTaxonomy < ImportHook
       def after_populate
         primary_ip = host.primary_interface.ip
+        primary_ip6 = host.primary_interface.ip6
 
         unless primary_ip
           logger.warn "Unable to assign subnet - reboot trigger may not be possible, primary interface is missing IP address"
           return
         end
 
         # set subnet
-        set_subnet(primary_ip)
+        set_subnets(primary_ip, primary_ip6)
         # set location and organization
         set_location
         set_organization
       end
 
       private
 
-      def set_subnet(ip)
-        host.primary_interface.subnet = suggested_subnet(ip)
+      def set_subnets(ip, ip6)
+        host.primary_interface.subnet = suggested_ipv4_subnet(ip)
+        subnet6 = suggested_ipv6_subnet(ip6) if ip6
+        logger.warn("Detected IPv6 subnet but it's not set for the host, unsupported") if subnet6
```

Comment:
Why is this unsupported? What if a host just has solely an IPv6 address?

Reply:
To do what with? We don't support IPv6 provisioning yet, excluding some special cases which are not PXE. On the other hand, it's reasonable to set it properly here. Changing.